lib/puppet/provider/cs_location/crm.rb

```diff
@@ -17,8 +17,30 @@
   # Decided to just go with relative.
   commands crm: 'crm'
 
+  # Path to the pacemakerd binary to check version for resource_discovery feature
+  # Decided to just go with relative.
+  commands pacemakerd: 'pacemakerd'
+
   mk_resource_methods
 
+  # rubocop:disable Lint/HandleExceptions
+  # we need to check if we run at least pacemakerd version 1.1.13 before enabling feature discovery
+  # see http://blog.clusterlabs.org/blog/2014/feature-spotlight-controllable-resource-discovery
+  begin
+    pacemakerd_version_string = pacemakerd('--version')
+    if pacemakerd_version_string
+      pacemakerd_version = pacemakerd_version_string.scan(%r{\d+\.\d+\.\d+}).first unless pacemakerd_version_string.nil?
+      if pacemakerd_version
+        if Puppet::Util::Package.versioncmp(pacemakerd_version, '1.1.13') >= 0
+          has_feature :discovery
+        end
+      end
+    end
+  rescue Puppet::MissingCommand
+    # on fresh systems pacemaker is not yet installed when facts are gathered, so we do just nothing, when the executable is missing
```

Comment:
we could do a debug log, so we don't have to disable that that lint

Reply:
Good idea, will also change that with the next push